hubblestack_nova/modules/cve_scan_v2.py

```diff
@@ -0,0 +1,283 @@
+                if page_num == 1:
+                    master_json = cve_json
+                    continue
+                
+                master_json = _build_json(master_json, cve_json)
+        
+        except Exception as exc: # Ask about error handling...
+            return
+
+        #Cache results.
+        try:
+            with open(cache_path, 'w') as cache_file:
+                json.dump(master_json, cache_file) 
+        except IOError as exc:
+            log.error('The results weren\'t able to be cached')
+                    
+    ret = {'Success':[], 'Failure':[]}   
+    
+    affected_pkgs = _get_cve_vulnerabilities(master_json)
+    local_pkgs = __salt__['pkg.list_pkgs'](versions_as_list=True)
+    
+    for pkg_obj in affected_pkgs:
+##TODO: for efficiency, should switch the loop to go through just the local_pkgs and check if it's in the affected packages.. 
+        if pkg_obj.get_pkg() in local_pkgs:
+            affected_version = pkg_obj.get_version()
+            for local_version in local_pkgs[pkg_obj.get_pkg()]:
+                if _is_vulnerable(local_version, affected_version, pkg_obj.get_operator()):
+                    ret['Failure'].append(pkg_obj.report())
+                else:
+                    ret['Success'].append(pkg_obj.get_pkg())
+        
+        else:
+            ret['Success'].append(pkg_obj.get_pkg())
+    return ret
+def _get_cve_vulnerabilities(query_results):
+    '''
+    Returns list of vulnerable package objects.
+    ### TODO ### return map of pkg->pkg_obj for more efficient loop structure
+    '''
+    
+    vulnerable_pkgs = []
+
+    # Make sure query was successful
+    if query_results['result'].lower() != 'ok':
+        return
+    
+    # Get os version to only add vulnerabilites that apply to local system 
+    osmajorrelease = str(__grains__['osrelease']).split('.')[0]
+
+    for report in query_results['data']['search']:
+        
+        #data:search
+        reporter = report['_source']['reporter']
+        cve_list = report['_source']['cvelist']
+        href = report['_source']['href']
+        score = report['_source']['cvss']['score']
+        
+        for pkg in report['_source']['affectedPackage']:
+            #data:search:_source:affectedPackages
+            
+            if pkg['OSVersion'] in ['any', osmajorrelease]: # Check if os version matches grains
+                pkg_obj = vulnerablePkg(pkg['packageName'], pkg['packageVersion'], score, pkg['operator'], reporter, href, cve_list)
+                vulnerable_pkgs.append(pkg_obj)   
+    return vulnerable_pkgs
+
+def _is_vulnerable(local_version, affected_version, operator):
+    '''
+    Given two version strings, and operator
+        returns whether the package is vulnerable or not.
+    '''
+    # Get rid of prefix if version number has one, ex '1:3.4.52'
+    if ':' in local_version:
+         local_version = local_version[local_version.index(':')+1:]
```

Comment:
another way to do this that might be cleaner:

```
_, _, local_version = local_version.partition(':')
```

The underscores are throwaway variables. Your version is correct, just could be cleaner.